### PR TITLE
Generate and publish user guide written in Asciidoc

### PR DESCRIPTION
**Pull request by bmuschko**
For more information see https://github.com/bmuschko/gradle-docker-plugin/issues/620. You can generate the documentation with the `asciidoctor` task. I'd be happy to roll in the release notes as well.

The guide contains the existing information from `README.asciidoc` and polishes/enhances it. The guide is published automatically to GitHub pages similar to the Groovydocs.

### CODE REVIEW

File: src/docs/asciidoc/40-about.adoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
+== About This Project
+
+=== Maintainers
+
+* https://github.com/cdancy[Christopher Dancy]
+* https://github.com/bmuschko[Benjamin Muschko]
+* https://github.com/orzeh[Orzeh]
```
Comment (bmuschko) [inline src/docs/asciidoc/40-about.adoc]: @orzeh Let me know if you'd like your real name here. Your GitHub account doesn't reflect it.

Comment (orzeh) [reply src/docs/asciidoc/40-about.adoc]: why not. My name is Łukasz Warchał

Comment (bmuschko) [reply src/docs/asciidoc/40-about.adoc]: done

**Comment by bmuschko** [comment]
We should cut a release soon after merging this to `master`. Alternatively, we could just publish the current state of documentation to GitHub pages.

**Comment by bmuschko** [comment]
@orzeh @cdancy Merged! Would you mind cutting a new release?

**Comment by cdancy** [comment]
@bmuschko I'll kick one now.

**Comment by bmuschko** [comment]
@cdancy I think you'll still need to add a date to the release notes but other than that they should be complete as far as I can tell.

**Comment by bmuschko** [comment]
@cdancy Looks like the user guide didn't publish properly. I think the path wasn't correct. I will have a look a little bit later today.

**Comment by cdancy** [comment]
Sounds good.

**Comment by bmuschko** [comment]
@cdancy The publishing issue has been fixed. GH Pages now has the latest version of the user guide online.

**Comment by cdancy** [comment]
@bmuschko thanks and yeah that looks really great.

**Pull request merged**